user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -4,554 +4,407 @@
+    /**
+     * Name of the secret containing the clients Certification Authority certificate.
+     */
+    public static final ConfigParameter<String> CA_CERT_SECRET_NAME = new ConfigParameter<>("STRIMZI_CA_CERT_SECRET_NAME", NON_EMPTY_STRING, "", true, CONFIG_VALUES);
+    /**
+     * Name of the secret containing the cluster Certification Authority certificate.
+     */
+    public static final ConfigParameter<String> CLUSTER_CA_CERT_SECRET_NAME = new ConfigParameter<>("STRIMZI_CLUSTER_CA_CERT_SECRET_NAME", STRING, null, false, CONFIG_VALUES);
+    /**
+     * Namespace with the CA secret.
+     */
+    public static final ConfigParameter<String> CA_NAMESPACE = new ConfigParameter<>("STRIMZI_CA_NAMESPACE", NON_EMPTY_STRING, null, false, CONFIG_VALUES);
+    /**
+     * The name of the secret containing the Entity User Operator key and certificate
+     */
+    public static final ConfigParameter<String> EO_KEY_SECRET_NAME = new ConfigParameter<>("STRIMZI_EO_KEY_SECRET_NAME", STRING, null, false, CONFIG_VALUES);
+    /**
+     * The name of the secret containing the clients Certification Authority key.
+     */
+    public static final ConfigParameter<String> CA_KEY_SECRET_NAME = new ConfigParameter<>("STRIMZI_CA_KEY_SECRET_NAME", NON_EMPTY_STRING, "", true, CONFIG_VALUES);
+    /**
+     * Map with labels which should be used to find the KafkaUser resources.
+     */
+    public static final ConfigParameter<Labels> LABELS = new ConfigParameter<>("STRIMZI_LABELS", LABEL_PREDICATE, "", true, CONFIG_VALUES);
+    /**
+     * How many milliseconds between reconciliation runs.
+     */
+    public static final ConfigParameter<Long> RECONCILIATION_INTERVAL_MS = new ConfigParameter<>("STRIMZI_FULL_RECONCILIATION_INTERVAL_MS", LONG, "120000", true, CONFIG_VALUES);
+    /**
+     * Kafka bootstrap servers list
+     */
+    public static final ConfigParameter<String> KAFKA_BOOTSTRAP_SERVERS = new ConfigParameter<>("STRIMZI_KAFKA_BOOTSTRAP_SERVERS", STRING, "localhost:9091", true, CONFIG_VALUES);
     /**
      * Configures the default prefix of user secrets created by the operator
      */
-    public static final String DEFAULT_SECRET_PREFIX = "";
-    static final int DEFAULT_SCRAM_SHA_PASSWORD_LENGTH = 32;
+    public static final ConfigParameter<String> SECRET_PREFIX = new ConfigParameter<>("STRIMZI_SECRET_PREFIX", STRING, "", false, CONFIG_VALUES);
+    /**
+     * Number of days for which the certificate should be valid
+     */
+    public static final ConfigParameter<Integer> CERTS_VALIDITY_DAYS = new ConfigParameter<>("STRIMZI_CLIENTS_CA_VALIDITY", INTEGER, "30", true, CONFIG_VALUES);
+    /**
+     * How long before the certificate expiration should the user certificate be renewed
+     */
+    public static final ConfigParameter<Integer> CERTS_RENEWAL_DAYS = new ConfigParameter<>("STRIMZI_CLIENTS_CA_RENEWAL", INTEGER, "365", true, CONFIG_VALUES);
```

Comment:
Can these be negative? I suspect we should force them to be strictly positive. One way to do that would be to write a method in `ConfigParameterParse` something like this:

```java
    static <T extends Number> ConfigParameterParser<T> strictlyPositive(ConfigParameterParser<T> parser) {
        return new ConfigParameterParser<T>() {
            public T parse(String s) { 
                var value = parser.parse(s);
                if (value.longValue() <= 0) {
                    throw new ...
                }
               return value;
            }
        };
    }
```

then you can do

```java
public static final ConfigParameter<Integer> CERTS_VALIDITY_DAYS = new ConfigParameter<>("STRIMZI_CLIENTS_CA_VALIDITY", strictlyPositive(INTEGER), "30", true, CONFIG_VALUES);
```

You could write similar factories methods for decorators which applied other validation (e.g. number in a range, string matching a regex), but we can add those on a case-by-case basis.